nest-forms-backend/src/utils/subservices/sharepoint.subservice.ts

```diff
@@ -0,0 +1,357 @@
+    const form = await this.prismaService.forms.findUnique({
+      where: {
+        id: formId,
+        archived: false,
+      },
+    })
+    if (form === null) {
+      throw this.throwerErrorGuard.NotFoundException(
+        FormsErrorsEnum.FORM_NOT_FOUND_ERROR,
+        FormsErrorsResponseEnum.FORM_NOT_FOUND_ERROR,
+      )
+    }
+
+    const formDefinition = getFormDefinitionBySlug(form.formDefinitionSlug)
+    if (!formDefinition) {
+      throw this.throwerErrorGuard.NotFoundException(
+        FormsErrorsEnum.FORM_DEFINITION_NOT_FOUND,
+        `${FormsErrorsResponseEnum.FORM_DEFINITION_NOT_FOUND} ${form.formDefinitionSlug}`,
+      )
+    }
+
+    if (
+      formDefinition.type !== FormDefinitionType.SlovenskoSkGeneric ||
+      !formDefinition.sharepointData
+    ) {
+      throw this.throwerErrorGuard.UnprocessableEntityException(
+        SharepointErrorsEnum.SHAREPOINT_DATA_NOT_PROVIDED,
+        `${SharepointErrorsResponseEnum.SHAREPOINT_DATA_NOT_PROVIDED} Form id: ${form.id}.`,
+      )
+    }
+
+    const accessToken = await this.getAccessToken()
+    const { sharepointData, schemas } = formDefinition
+    const formTitle = this.getTitle(form, formDefinition)
+    const fields = await this.mapColumnsToFields(
+      Object.keys(sharepointData.columnMap),
+      accessToken,
+      sharepointData.databaseName,
+    )
+    const jsonDataExtraDataOmitted = omitExtraData(
+      schemas.schema,
+      form.formDataJson as GenericObjectType,
+    )
+    const valuesForFields = getValuesForFields(
+      sharepointData,
+      { ...form, title: formTitle },
+      jsonDataExtraDataOmitted,
+      fields,
+    )
+
+    if (sharepointData.oneToOne) {
+      const oneToOneOriginalTableFields = await this.mapColumnsToFields(
+        Object.values(sharepointData.oneToOne).map(
+          (record) => record.originalTableId,
+        ),
+        accessToken,
+        sharepointData.databaseName,
+      )
+
+      await Promise.all(
+        Object.values(sharepointData.oneToOne).map(async (value) => {
+          const oneToOneFields = await this.mapColumnsToFields(
+            Object.keys(value.columnMap),
+            accessToken,
+            value.databaseName,
+          )
+          const valuesForFieldsOneToOne = getValuesForFields(
+            value,
+            { ...form, title: formTitle },
+            jsonDataExtraDataOmitted,
+            oneToOneFields,
+          )
+
+          const addedId = await this.postDataToSharepoint(
+            value.databaseName,
+            accessToken,
+            valuesForFieldsOneToOne,
+          )
+
+          valuesForFields[
+            `${oneToOneOriginalTableFields[value.originalTableId]}Id`
+          ] = addedId
+        }),
+      )
+    }
+
+    const baseId = await this.postDataToSharepoint(
+      sharepointData.databaseName,
+      accessToken,
+      valuesForFields,
+    )
+
+    if (sharepointData.oneToMany) {
+      await Promise.all(
```

Comment:
Not sure about the nested `Promise.all` I would personaly create array of arrays, then use [Array#flat()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flat) it and then await it.

Reply:
Like this https://chatkit.app/s/jyeun5okf1ugiud

Reply:
good point, I have changed this nested promise